lib/cdo/shared_constants.rb

```diff
@@ -0,0 +1,29 @@
+# This file is generated via generateSharedConstants.rb. DO NOT CHANGE MANUALLY
+
+module SharedConstants
+  # Used to communicate different types of levels
+  LevelKind = OpenStruct.new(
+    {
+      "peer_review": "peer_review",
+      "assessment": "assessment",
+      "puzzle": "puzzle",
+      "unplugged": "unplugged",
+      "level": "level"
+    }
+  )
+
+  # Different possibilities for level.status, used to communicate how user has performed on a given level
+  LevelStatus = OpenStruct.new(
+    {
+      "not_tried": "not_tried",
+      "submitted": "submitted",
+      "locked": "locked",
+      "perfect": "perfect",
+      "passed": "passed",
+      "attempted": "attempted",
+      "review_accepted": "review_accepted",
+      "review_rejected": "review_rejected",
+      "dots_disabled": "dots_disabled"
+    }
+  )
```

Comment:
Any reason why we're going against ruby style guidelines here? Why not produce these hashes with symbol keys instead of string?

It can be done by writing it directly as, e.g. `not_tried: "not_tried"`, or constructing the entire hash as-is, then calling `.symbolize_keys`

The variable names also, in typical Ruby [style](https://github.com/bbatsov/ruby-style-guide#snake-case-symbols-methods-vars), should start lower case, e.g. `levelStatus`.